lib/instrument/headers.js

```diff
@@ -0,0 +1,339 @@
+    // eslint-disable-next-line no-undef
+    buf.writeBigUInt64LE(BigInt(value), 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+
+  appendSignedInt (value) {
+    const buf = Buffer.alloc(12);
+    buf.writeUInt32LE(10, 0);
+    buf.writeUInt32LE(3, 4);
+    buf.writeInt32LE(value, 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+  }
+
+  appendSignedLong (value) {
+    const buf = Buffer.alloc(16);
+    this._buf.writeUInt32LE(10, 0);
+    this._buf.writeUInt32LE(6, 4);
+    // eslint-disable-next-line no-undef
+    this._buf.writeBigInt64LE(BigInt(value), 8);
+    this._buf = Buffer.concat([this._buf, buf]);
+
+  }
+
+  appendObject (value) {
+    const buf = archive(value);
+    const buf2 = Buffer.alloc(12);
+    buf2.writeUInt32LE(10, 0);
+    buf2.writeUInt32LE(2, 4);
+    buf2.writeUInt32LE(buf.length, 8);
+    this._buf = Buffer.concat([this._buf, buf2, buf]);
+
+  }
+
+  getBytes () {
+    const out = Buffer.alloc(16);
+    out.writeUInt32LE(DTX_MESSAGE_AUX_HEADER, 0);
+    out.writeUInt32LE(this._buf.length, 8);
+    return Buffer.concat([out, this._buf]);
+  }
+}
+
+
+class DTXMessage {
+  constructor (opts = {}) {
+    const {
+      identifier,
+      channelCode,
+      selector,
+      expectsReply,
+      conversationIndex = 0,
+      flags = FLAG_TYPES.send
+    } = opts;
+    this._messageHeader = undefined;
+    this._payloadHeader = undefined;
+    this._identifier = identifier;
+    this._channelCode = channelCode;
+    this._expectsReply = expectsReply;
+    this._selector = selector;
+    this._conversationIndex = conversationIndex;
+    this._flags = flags;
+    this.auxiliaries = new DTXMessageAux();
+  }
+  get identifier () {
+    return this._messageHeader?.identifier;
+  }
+  get channelCode () {
+    return this._messageHeader?.channel;
+  }
+  get expectsReply () {
+    return this._messageHeader?.expectsReply;
+  }
+  get selector () {
+    return this._selector;
+  }
+  set selector (data) {
+    this._selector = data;
+  }
+  get conversationIndex () {
+    return this._messageHeader?.conversationIndex;
+  }
+
+  /**
+   * DTXMessage Buffer: DTXMessageHeader + PayloadHeader + DTXMessageAuxBuffer(nullable) + selector(nullable)
+   * @returns {Buffer}
+   */
+  build () {
+    const sel = this._selector ? archive(this._selector) : Buffer.alloc(0);
+    const aux = this.auxiliaries.build();
+    const payloadHeader = DTXMessagePayloadHeader.build({
+      flags: this._flags,
+      auxLength: aux.length,
+      totalLength: aux.length + sel.length,
+    });
+
+    const messageHeader = DTXMessageHeader.build({
+      payloadLength: DTX_MESSAGE_PAYLOAD_HEADER_LENGTH + aux.length + sel.length,
+      identifier: this._identifier,
+      conversationIndex: this._conversationIndex,
+      channel: this._channelCode,
+      expectsReply: this._expectsReply ? 1 : 0
+    });
+    return Buffer.concat([messageHeader, payloadHeader, aux, sel]);
+  }
+
+  /**
+   * @param headerBuf
```

Comment:
Would it be possible to also describe argument types and purpose like in other modules?